### PR TITLE
feat(data-warehouse): Added Chargebee data source

### PR DESCRIPTION
**Pull request by rossgray**
## Problem

We want to be able to import Chargebee data.

## Changes

Adds a new data source for Chargebee.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Should do.

## How did you test this code?

Added unit test and tested importing data locally using a Chargebee sandbox account.

### CODE REVIEW

File: posthog/warehouse/api/external_data_source.py [hunk 1]
```diff
@@ -834,6 +873,15 @@ def database_schema(self, request: Request, *arg: Any, **kwargs: Any):
...
             return Response(status=status.HTTP_200_OK, data=result_mapped_to_options)
+        elif source_type == ExternalDataSource.Type.CHARGEBEE:
+            api_key = request.data.get("api_key", "")
+            site_name = request.data.get("site_name", "")
+
+            if not validate_chargebee_credentials(api_key=api_key, site_name=site_name):
+                return Response(
+                    status=status.HTTP_400_BAD_REQUEST,
+                    data={"message": "Invalid credentials: Chargebee credentials are incorrect"},
+                )
```
Comment (Gilbert09) [inline posthog/warehouse/api/external_data_source.py:884]: Nice, we also wanna check the subdomain to ensure users can't do any RCE attacks here, see https://github.com/PostHog/posthog/blob/master/posthog/warehouse/api/external_data_source.py#L781-L786

File: posthog/temporal/data_imports/pipelines/chargebee/__init__.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+from posthog.temporal.data_imports.pipelines.rest_source import (
+    RESTAPIConfig,
+    rest_api_resources,
+)
+from posthog.temporal.data_imports.pipelines.rest_source.typing import EndpointResource
+
+
+def get_resource(name: str, is_incremental: bool) -> EndpointResource:
+    resources: dict[str, EndpointResource] = {
+        "Customers": {
```
Comment (Gilbert09) [inline posthog/temporal/data_imports/pipelines/chargebee/__init__.py:18]: I reckon starting with customers, subscriptions, orders, invoices, transactions would be good

**Review by Gilbert09 — CHANGES_REQUESTED** [review]
One change, just to check the subdomain, otherwise this looks good!

File: posthog/warehouse/api/external_data_source.py (hunk 1 above)
Comment (rossgray) [reply posthog/warehouse/api/external_data_source.py:884]: ah yes, I missed that, thanks!

**Review by Gilbert09 — APPROVED** [review]

**Pull request merged**